apps/pay/app/ssr-client.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpLink } from "@apollo/client"
+import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc"
+import {
+  NextSSRApolloClient,
+  NextSSRInMemoryCache,
+} from "@apollo/experimental-nextjs-app-support/ssr"
+
+import { getClientSideGqlConfig } from "@/config/config"
+
+export const { getClient } = registerApolloClient(() => {
+  return new NextSSRApolloClient({
+    cache: new NextSSRInMemoryCache(),
+    link: new HttpLink({
+      uri: getClientSideGqlConfig().coreGqlUrl,
+      fetchOptions: { cache: "no-store" },
+
+      // TODO uncomment this we start using otel
+      //   fetch: (uri, options) => {
+      //     const headersWithTrace = options?.headers || {}
+      //     propagation.inject(context.active(), headersWithTrace)
+      //     return fetch(String(uri), {
+      //       ...options,
+      //       headers: headersWithTrace,
+      //     })
+      //   },
```

Comment:
```suggestion
      // TODO uncomment this we start using otel
      //   fetch: (uri, options) => {
      //     const headersWithTrace = options?.headers || {}
      //     propagation.inject(context.active(), headersWithTrace)
      //     return fetch(String(uri), {
      //       ...options,
      //       headers: headersWithTrace,
      //     })
      //   },
```

not sure what is this comment for. maybe this was a TODO? can't we have otel for the node tasks?

Reply:
Yes, this was a to-do.